app/lib/actions/katello/capsule_content/sync.rb

```diff
@@ -38,13 +38,24 @@ def plan(smart_proxy, options = {})
           affected_repos = affected_repositories(capsule_content, environment, content_view, repository)
           need_updates = repos_needing_updates(capsule_content, affected_repos)
           repository_ids = get_repository_ids(capsule_content, environment, content_view, repository)
+
+          # Create a list of non-puppet repos to sync first, and then sync puppet repos last.
+          # Puppet modules may refer to non-puppet content in the same CV, and thus need to publish last.
+          puppet_repository_ids = []
+          non_puppet_repository_ids = []
+          repository_ids.each do |repo_id|
+            repo = ::Katello::Repository.find_by(pulp_id: repo_id) || ::Katello::ContentViewPuppetEnvironment.find_by(pulp_id: repo_id)
```

Comment:
could we do something like this (finding a repo or CVPE) on the Repository model level? I feel like I have seen similar logic in other places.